### PR TITLE
Optimize generate_deployments_for_service, taking it from ~3 minutes …

### PR DESCRIPTION
**Pull request by EvanKrall**
…on yelp-main to ~20 seconds (most of which is waiting for git)

PAASTA-16665

### CODE REVIEW

File: paasta_tools/utils.py [hunk 1]
```diff
@@ -2852,6 +2852,12 @@ def load_service_instance_configs(
     return merged
 
 
+# be extremely sure you do not modify the return value of this.
```
Comment (qui) [inline paasta_tools/utils.py:2855]: Hm, how much faster does this make things? service_configuration_lib already does some YAML caching:  https://github.com/Yelp/service_configuration_lib/blob/master/service_configuration_lib/__init__.py#L90

But it always deep-copies, which makes it a little slower. Just wondering if the speed is worth having to worry about accidental mutations here.

Comment (EvanKrall) [reply paasta_tools/utils.py:2855]: Yeah, the deep copy is the second-slowest part. In my pyflame of stock `generate_deployments_for_service`, copy.deepcopy takes 15.5% of time, so roughly 25-30s. (`get_deploy_group_mappings` is the bulk of the rest of it)

Comment (keymone) [reply paasta_tools/utils.py:2855]: seems safe to do pickle+unpickle? should be faster iirc

Comment (EvanKrall) [reply paasta_tools/utils.py:2855]: really? it would surprise me if serializing/deserializing is faster than deepcopy.

**Pull request merged**